homeassistant/helpers/dispatcher.py

```diff
@@ -236,4 +243,13 @@ def async_dispatcher_send_internal[*_Ts](
         if job is None:
             job = _generate_job(signal, target)
             target_list[target] = job
-        hass.async_run_hass_job(job, *args)
+        # We do not wrap Callback jobs in catch_log_exception since
+        # single use dispatchers spend more time wrapping the callback
+        # than the actual callback takes to run in many cases.
+        if job.job_type is HassJobType.Callback:
```

Comment:
I was really hoping to not add a branch here but given a callback is the common case I think I can be ok with it.